mage/query-modules/python/gnn-node-classification.md

```diff
@@ -0,0 +1,342 @@
+
+### About the query module
+
+This query module contains all necessary functions user should need to train GNN model on Memgraph. User first fills Memgraph nodes and edges, either with prepared dataset or made by its own. 
+
+In this query module these are some of functionalities that can be used:
+
+ 1. **graph can be heterogeneous or homogeneous**
+    - multiple node and edge labels are allowed
+ 2. **no restrictions on dataset size** 
+    - implemented data loader
+ 3.  **multiple model architectures** to work on (inductive learning)
+    - GAT with Jumping Knowledge
+    - multiple version of GAT
+    - GraphSAGE
+ 4. model **training**
+    1. saving of last $n$ models
+    2. early stopping
+    3. metrics calculations
+ 5. **predict** on one node
+ 6. **saving** models
+ 7. **loading** models
+
+
+If you want to explore our implementation, jump to
+**[github/memgraph/mage](https://github.com/memgraph/mage)** and find
+`python/node_classification.py`. You can also jump to the [download
+page](https://memgraph.com/download), download **Memgraph Platform** and fire up
+**Node Classification**. We have prepared an homogenouse graph **Yelp-Fraud (Multi-relational Graph Dataset for Yelp Spam Review Detection)** dataset on which you can
+explore node classification using a **[Jupyter
+Notebook](https://github.com/memgraph/jupyter-memgraph-tutorials)** and other Memgraph **home-made** heterogeneous graph fraud detection dataset.
+
+Feel free to open a **[GitHub issue](https://github.com/memgraph/mage/issues)**
+or start a discussion on **[Discord](https://discord.gg/memgraph)** if you want
+to speed up development.
+
+How should **you** use the following module? Load dataset, call `set_model_parameters`, and let the training begin! When training is finished, models are saved. You can test them on other data (which model has not already seen!) and check the results!
+The module reports the [mean average
+precision](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html)
+for every batch _training_ or _evaluation_ done.
+
+### Usage
+
+The following procedure is expected when using **Node classification**:
+
+- fill Memgraph with data using Cypher, CSV, or easily with our already prepared datasets
+- you should first set parameters by calling `set_model_parameters()` function
+  - you should probably set **node features** label
+- now call `train()` function
+- take a look of training data with `get_training_data()` function
+- optionally use `save_model()` and `load_model()` if you want
+- use `predict()` for predicting classes on **not already seen** nodes
+
+
+
+
+:::info
+
+This **MAGE** module is still in its early stage. We intend to use it only for
+**learning** activities. If you wish to make it production-ready, make sure
```

Comment:
Furthermore, I’d change the message here:
* "If you wish to make it production-ready" < it's our job to do that, not the user's
* Asking the reader to give us a star right after telling them this is not a finished feature seems like it would backfire

Reply:
I don't know if it will backfire, but this way we explain to users that intention of this module is still to learn and explore about node classification